px-demo-snippet.html

```diff
@@ -203,26 +202,292 @@
+        this.$.jeditor.innerHTML = highlighted;
+        return;
+      }
+
+      // Otherwise, send the code for highlighting
+      highlighted = this._fireOutputEvent(this._output, 'html');
+
+      // If prism highlighter hasn't loaded yet, try ourselves again 50ms later
+      if ((typeof highlighted !== 'string') || highlighted.trim().substr(0,5) !== "<span") {
+        this.async(this._render, 50);
+        return;
+      }
+
+      var grouped = this.noIndentation ? highlighted : this._applyGroups(highlighted);
+      var scoped = this._applyStyleScope(grouped);
+
+      // If we get here, the code is highlighted. Send it to the cache and set it.
+      this.__highlightCache[this._output] = scoped;
+      this.$.jeditor.innerHTML = scoped;
+    },
+
+    /**
+     * Wrap our highlighter output in divs to allow each attr/value to be on its own line
+     *
+     * returns new html string with divs
+     */
+    _applyGroups: function(html) {
+      var attrs = /(<span class="token attr-name">)/g,  //search for each attr name span
+          openTag = /(<span class="token punctuation">><\/span><\/span>)\s*(<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;<\/span>)/g, //search for html tag
+          closingTag = /(<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;\/)/g, //search for the end closing tag
+          commentstart = /(<span class="token comment" spellcheck="true">)/g, //I think just for px-tab
+          commentend = /(--><\/span>)/g, //I think just for px-tab
+          lazy = /<\/div>/,  //probably a better way to remove that first one we add, but too lazy
+          str = html.replace(attrs, '</div><div class="keyVal">$1')
+                    .replace(openTag, '$1<div class="openTag"><div>$2')
+                    .replace(closingTag, '</div></div>$1')
+                    .replace(commentstart, '<div class="comment">$1')
+                    .replace(commentend, '$1</div>')
+                    .replace(lazy,'') + '</div>';
+
+      // honestly not sure why this needs to be separate
+      // Prism marks up style attrs a bit differently and does a dumb search for style.
+      // if you have an attr, feature-style for example, it gets the other markup causing the regex to add breaks on the style
+      // this removes the added divs for this case
+      var style = /((<span class="token style-attr language-css">)(<\/div><div class="keyVal"><span class="token attr-name"><\/div><div class="keyVal">))/g,
+          str2 = str.replace(style, '');
+
+      return str2;
+    },
+
+    /**
+     * Hack to apply style scope to our highlighter generated code so we can apply styles
+     *
+     * return highlighter input with style scope added
+     */
+    _applyStyleScope: function(html) {
+      if(Polymer.Settings.useNativeShadow) { return html; }
+
+      var re = /(class="[^"]+)/g,
+          str = html.replace(re, '$1 style-scope px-demo-snippet');
```

Comment:
You're not sure you'll be the only `px-demo-snippet` on the page. If Polymer stamps more than one the last class will not be `px-demo-snippet`, it'll be `px-demo-snippet-n` where ni is the number of snippet elements stamped so far. You need to read this dynamically in case you're px-demo-snippet-1 or px-demo-snippet-8 or whatever number.